src/main/java/io/jenkins/plugins/artifact_manager_jclouds/JCloudsArtifactManager.java

```diff
@@ -132,7 +132,10 @@ public void archive(FilePath workspace, Launcher launcher, BuildListener listene
             String blobPath = getBlobPath(path);
             Blob blob = blobStore.blobBuilder(blobPath).build();
             blob.getMetadata().setContainer(provider.getContainer());
-            blob.getMetadata().getContentMetadata().setContentType(contentTypes.get(entry.getKey()));
+            String contentType = contentTypes.get(entry.getKey());
+            if (contentType != null) {
+                blob.getMetadata().getContentMetadata().setContentType(contentType);
```

Comment:
I do not know if this data is required in all cases or if it may be omitted in some cases as I have done here.

Reply:
Judging by #107 it is probably OK, though unclear. You can always fall back to `application/octet-stream`.

Reply:
From some quick testing, `content-type:` seems to be included in the data that gets signed even when there is no detected content type and this line does not execute, so I will fall back to "application/octet-stream".

Reply:
See https://github.com/jenkinsci/artifact-manager-s3-plugin/pull/349/commits/a68f97f5e84a3a8d5bb0a0acc7dde34be682b74d.